### PR TITLE
Accounts

### PR DESCRIPTION
**Pull request by ramirez915**
I got the accounts tests passing and it seems that I managed to get the database properly updating after the contract has been validated. Only thing omitted was the comparison of structs because I was unable to figure out how to use the cmp.Equal() along with the AllowUnexported(). Also I noticed that the updated database is currently not being tested for.

### CODE REVIEW

**Comment by ramirez915** [comment]
Alright cool I'll get on it. I left the prints for now so that you can see that things are being updated properly since there wasn't a test for it at the moment. I forgot what I did that caused the table to remain but I will try to look for the issue.

**Comment by kstola2** [comment]
@ramirez915 After doing some research, I don't think that using `reflect.DeepEqual` will be a big problem. Passing an ignore flag doesn't solve the problem of still needing to compare that field, so we would need to `reflect.DeepEqual` on it anyway after `cmp.Equal`ing the other fields. So I think it's alright to uncomment that comparison block in the test and just use `reflect.DeepEqual` instead.

**Comment by ramirez915** [comment]
I cleaned up the code and substituted the `cmp.Equal` function for the `reflect.DeepEqual` function. The database file seems to be removed after each run.

File: producer/src/accounts/accounts.go [hunk 1]
```diff
@@ -0,0 +1,243 @@
...
+increase value of recipient public key hash account by contract value
+*/
+func (c *Contract) UpdateAccountBalanceTable(table string) {
+	tbl, err := sql.Open("sqlite3", table)
+	if err != nil {
+		fmt.Println("Failed to open sqlite3 table")
+	}
+	defer tbl.Close()
+
+	rows, err := tbl.Query("SELECT public_key_hash , balance, nonce FROM account_balances")
```
Comment (kstola2) [inline producer/src/accounts/accounts.go]: Can you use a `WHERE public_key_hash = ?` here so that you won't need to loop through each item later?

**Comment by bashbashbash** [comment]
What's the status of this PR?

**Comment by kstola2** [comment]
@bashbashbash the status is you and me are working on it

File: producer/src/accounts/accounts.go [hunk 2]
```diff
@@ -0,0 +1,269 @@
...
+	Version         uint16
+	SenderPubKey    ecdsa.PublicKey
+	SigLen          uint8  // len of the signature
+	Signature       []byte // size varies
+	RecipPubKeyHash []byte // 32 bytes
+	Value           uint64
+	Nonce           uint64
+}
+
+const TableName string = "account_balances"
```
Comment (kstola2) [inline producer/src/accounts/accounts.go]: What is the purpose of this?

**Pull request merged**